### DISCUSSION TITLE
Controlling the locking mechanism?

### DISCUSSION THREAD

**Discussion by and1tran** [MatthiasValvekens/pyHanko#250] [Q&A] [answered]
Hi Matthias,

I have signed a PDF with pyHanko and it works fantastically. Now I need to pass it to my supervisor to also sign but signing and editing is locked. Is there a way to control whether the pdf is locked to signing only or locked from just editing?

**Comment by MatthiasValvekens** [comment, marked as answer]
Hi @and1tran,


Unless you ask for it when creating the signature, pyHanko will not set any _explicit_ locks. Exactly what kind of incremental changes are still allowed after signing depends on the viewer/validator, but adding new signatures is typically not a problem.

To reduce the potential for confusion/mistakes/operator error, you could also set up two signature fields prior to signing, use one for your own signature and let your supervisor sign the other one. Slightly more complicated, but can make things easier down the road.


Any chance that your supervisor is trying to add a "drawn" (non-cryptographic) signature? Because many PDF viewers will not allow you to doodle on top of a signed PDF. If that's not it, then you'll have to share the CLI invocations and/or Python code you used to add your signature.
